src/Ocean/SuperModels.jl

```diff
@@ -84,6 +84,10 @@ function HydrostaticBoussinesqSuperModel(;
     filters = nothing,
     modeldata = NamedTuple(),
     init_on_cpu = true,
+    boundary_conditions = (
```

Comment:
@glwagner is there a reason to have some defaults here, instead of

```
boundary_conditions = (),
```

Reply:
It depends on what we want the default model to be. If our default domain / geometry has boundaries, I think we also need to include default boundary conditions somewhere, somehow.

I think we can make specifying boundary conditions a bit easier for most of the problems we are interested in by constraining the input a bit. We can generally use the terms "east, west, north, south, top, bottom" to refer to the 6 end points of a domain (2 for each of the 3 directions). This works for simple `CartesianDomain`s and `SphericalDomain`s. More complicated boundary conditions for unstructured meshes would require the more flexible interface that boundary condition "tag"s permits, so we can keep that as a fallback when we need it, too.

Reply:
Definitely BC abstractions (and implementation of current abstraction) need work.

I think default of periodic in X and Y makes sense, not sure about Z. Periodic is nice for symmetry. 
Z as periodic by default its a bit too obscure turbulence/CFD for most people. Most ocean users would expect there to be an ocean floor or an open boundary. Also with g it doesn't usually work very well, things fall out of the bottom......

Reply:
Doesn't our balance law sort of require there to be a free surface at the top? Does vertically periodic even make sense for us?

Reply:
Vertically periodic does not make sense for the model - I was thinking more topologically. At the model level it doesn't. I think periodic in X and Y is a good default though. For vertical we need to choose between NoSlip and FreeSlip at bottom for Kinematic - both are legitimate defaults. For Surface, something inert e.g. NoSlip, NoFlux etc... would be my preference.

Reply:
@jm-c any preference on default BC's?

Reply:
I think no flux boundary conditions for solid boundaries make the most sense because they simplest model from the standpoint of the numerics. A NoSlip might not be inert, since the flux of momentum associated with it must be modeled by a flux reconstruction of some kind. 

In addition, NoSlip boundary conditions are problematic because they require a certain amount of viscous friction and, depending on the numerical methods used, may require resolving a boundary layer of unknown thickness. Thus as a default this can create problems for users who just want to spin up some simple initial condition to start out.

For some problems (depending on numerics) we can run without viscosity at all. This might be the simplest of all defaults --- numerical diffusion only with free slip and insulating conditions at solid boundaries.

Reply:
That sounds good to me. I think going with default as the "simplest sensible thing possible, but no simpler" is a good principle. "Simplest sensible thing possible" is of course a little subjective, but that seems OK - it allows some opinionated element into the mix to guide people.

Reply:
I agree with you @glwagner , this default BC makes sense. It's also the type of default BC that is/was there in src/Ocean/OceanBC.jl if I am right.